vsintegration/src/FSharp.Editor/Navigation/GoToDefinition.fs

```diff
@@ -706,10 +713,17 @@ type internal FSharpNavigation(metadataAsSource: FSharpMetadataAsSourceService,
             let gtd = GoToDefinition(metadataAsSource)
             let! result = gtd.FindDefinitionAtPosition(initialDoc, position)
 
-            return
-                match result with
-                | ValueSome(FSharpGoToDefinitionResult.NavigableItem(navItem), _) -> ImmutableArray.create navItem
-                | _ -> ImmutableArray.empty
+            match result with
+            | ValueSome(FSharpGoToDefinitionResult.NavigableItem(navItem), _) -> return ImmutableArray.create navItem
+            | ValueSome(FSharpGoToDefinitionResult.ExternalAssembly(targetSymbolUse, metadataReferences), _) ->
+                let! res = gtd.NavigateToExternalDeclarationAsync(targetSymbolUse, metadataReferences)
+
+                if not res then
+                    return ImmutableArray.empty
+                else
+                    // TODO: return something? Maybe cancelled task?
```

Comment:
I don't think cancelled task would be any better here.

Reply:
I'm not sure tbh what would be a good return value here.

Reply:
The empty immarray feels appropriate.
It probably an errornous situation, but might as well come from stale data.

Idea:
Send this situation to telemetry reporter.

Reply:
> The empty immarray feels appropriate.
> 

Empty array will lead to message which will say something like "failed to find symbol", or it leads now.

> It probably is an errornous situation, but might as well come from stale data.
> 

Not really. We return empty array when we didn't find anything. In case of external navigation, we need to handle it ourselves and somehow tell Roslyn "nothing found but it's fine", that's why I thought of cancelled task - we cancel Roslyn navigation and deal with it ourselves.

Reply:
Oh you are right, I misread to which branch this belongs.
I meant to log telementry in case we did not find anything ourselves.

If we did find it, cancelling it seems like a good approach, indeed.